scipy/spatial/transform/_rotation.pyx

```diff
@@ -2124,13 +2171,93 @@ cdef class Rotation:
         cdef double[:] angles = _empty1(num_rotations)
 
         for ind in range(num_rotations):
-            angles[ind] = 2 * atan2(_norm3(quat[ind, :3]), abs(quat[ind, 3]))
+            angles[ind] = _quat_magnitude(quat[ind])
 
         if self._single:
             return angles[0]
         else:
             return np.asarray(angles)
 
+    @cython.embedsignature(True)
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    def angle_to(Rotation self, Rotation other):
+        """Get the smallest angular distance(s) to another rotation.
+
+        For two rotations ``p`` and ``q``, the angular distance
+        ``p.angle_to(q)`` is equivalent to calculating
+        ``(q * p.inv()).magnitude()``.
+
+        Parameters
+        ----------
+        other : `Rotation` instance
+            Object containing the rotations to measure the angle to.
+
+        Returns
+        -------
+        angle : ndarray or float
+            Angle(s) in radians, float if object contains a single rotation
+            and ndarray if object contains multiple rotations. The angle
+            will always be in the range [0, pi].
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+        >>> import numpy as np
+        >>> p = R.from_quat([0, 0, 0, 1])
+        >>> q = R.from_quat(np.eye(4))
+        >>> p.angle_to(q)
+        array([3.14159265, 3.14159265, 3.14159265, 0.        ])
+
+        Angle to a single rotation:
+
+        >>> p.angle_to(q[0])
+        3.141592653589793
+        """
+        return self.rotation_to(other).magnitude()
+
+
+    @cython.embedsignature(True)
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    def approx_equal(Rotation self, Rotation other, float atol=1e-12):
+        """Determine if another rotation is approximately equal to this one.
```

Comment:
I don't think we have (m)any approx functions in the regular name spaces. If anything this seems like a function for testing, or - given the simplicity of the implementation - something that people can do themselves. This would also absolve us of having to choose a "reasonable" tolerance default.

Reply:
Given the confusion in the linked issue and the conversations about the double cover of rotation space, I think the operation here while simple, is very non-obvious to many people. The default behavior I’ve seen people do is check for approximate equality between one of the representations, all of which have more or less common edge cases which will fail (unexpectedly). So I’m open to a name change or a better default tolerance but I do think this is useful to include.

Reply:
You can’t directly compare quaternions because of the double cover, and you can’t check rotation vectors for the same reason. I think rotation matrices should be fine to compare directly, but it’s a nonlinear mapping between the tolerance of those matrix entries and an angular tolerance.

Reply:
Sure, but once there's the `angle_to` function that this PR introduces, then it's as simple as:
```
p.angle_to(q) < my_tol
```

So definitely I'm in favour of `angle_to`, but not necessarily of such a trivial derivative. And the presence of an approx function does not mean people will not do the wrong thing. Adding stuff to the API has a maintenance cost long into the future, so introducing a wrapper for a one-liner (_after_ `angle_to` is available) would really need very strong reasons IMO.

Reply:
The short implementation doesn't bother me. Having a prominent "this is the one right way to do this common, but subtle and often incorrectly-implemented, task" is worthwhile, IMO. It's not so much a wrapper for a one-liner as an important function that happens to have a short implementation.

Reply:
If you really insisted on only having the budget for one additional method, I'd rather have `approx_equal()` than `angle_to()`, TBH.

Reply:
> If you really insisted on only having the budget for one additional method, I'd rather have `approx_equal()` than `angle_to()`, TBH.

I'm not going to insist. I don't know this side of the code-base well-enough, and was only arguing from general principles. I'm fine with it if you think it's worthwhile.

Reply:
My personal perspective here is that `approx_equal` is the important function to add. The other ones came along for free, but if there’s a strong desire to keep the API small I can remove them.